compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/Magnifier.uikit.kt

```diff
@@ -0,0 +1,350 @@
+        val view = view ?: return
+        val density = density ?: return
+        magnifier = platformMagnifierFactory.create(
+            view = view,
+            density = density,
+            color = color
+        )
+        updateSizeIfNecessary()
+    }
+
+    private fun updateMagnifier() {
+        val magnifier = magnifier ?: return
+        val density = density ?: return
+
+        val sourceCenterOffset = sourceCenter(density)
+        sourceCenterInRoot =
+            if (anchorPositionInRoot.isSpecified && sourceCenterOffset.isSpecified) {
+                anchorPositionInRoot + sourceCenterOffset
+            } else {
+                Offset.Unspecified
+            }
+
+        // Once the position is set, it's never null again, so we don't need to worry
+        // about dismissing the magnifier if this expression changes value.
+        if (sourceCenterInRoot.isSpecified) {
+            // Calculate magnifier center if it's provided. Only accept if the returned value is
+            // specified. Then add [anchorPositionInRoot] for relative positioning.
+            val magnifierCenter = magnifierCenter?.invoke(density)
+                ?.takeIf { it.isSpecified }
+                ?.let { anchorPositionInRoot + it }
+                ?: Offset.Unspecified
+
+            magnifier.update(
+                sourceCenter = sourceCenterInRoot,
+                magnifierCenter = magnifierCenter,
+            )
+            updateSizeIfNecessary()
+        } else {
+            // Can't place the magnifier at an unspecified location, so just hide it.
+            magnifier.dismiss()
+        }
+    }
+
+    private fun updateSizeIfNecessary() {
+        val magnifier = magnifier ?: return
+        val density = density ?: return
+
+        if (magnifier.size != previousSize) {
+            onSizeChanged?.invoke(with(density) { magnifier.size.toSize().toDpSize() })
+            previousSize = magnifier.size
+        }
+    }
+
+    override fun ContentDrawScope.draw() {
+        drawContent()
+        // don't update the magnifier immediately, actual frame draw happens right after all draw
+        // commands are recorded. Magnifier update should happen in the next frame.
+        coroutineScope.launch {
+            withFrameMillis { }
+            magnifier?.updateContent()
+        }
+    }
+
+    override fun onGloballyPositioned(coordinates: LayoutCoordinates) {
+        // The mutable state must store the Offset, not the LocalCoordinates, because the same
+        // LocalCoordinates instance may be sent to this callback multiple times, not implement
+        // equals, or be stable, and so won't invalidate the snapshotFlow.
+        anchorPositionInRoot = coordinates.positionInRoot()
+    }
+
+    override fun SemanticsPropertyReceiver.applySemantics() {
+        this[MagnifierPositionInRoot] = { sourceCenterInRoot }
+    }
+}
+
+internal fun isPlatformMagnifierSupported() =
+    available(OS.Ios to OSVersion(major = 17))
```

Comment:
There is a check for iOS version. It is implemented incorrectly?

Reply:
Yes, the problem is due to linking. K/N can see the symbol and tried to link it during the launch.

Reply:
I'll promote this problem to K/N team, the only known workaround as of now is to use strings and C-apis directly to avoid mentioning the symbol altogether.

Reply:
>use strings and C-apis directly

Do you mean using cinterop like [here ](https://github.com/JetBrains/compose-multiplatform-core/tree/jb-main/compose/ui/ui-uikit/src/uikitMain/objc/CMPUIKitUtils) or something different?

Reply:
> simply hide the symbols from cinterop

I also thought about it. Nice solution. I seen some places in skiko where iOS 13 is checked the same way and class is loaded. Should it be fixed as well?

Reply:
Methods are not a big deal, ObjC will simply throw unrecognised selector exception. The classes are a problem (currently), can you leave a link?